src/index.js

```diff
@@ -26,6 +26,18 @@ const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));
 
 export async function run() {
   try {
+    // custom working directory
+    const workingDirectoryInput = core.getInput('working-directory');
+    if (workingDirectoryInput) {
+      core.info(`Starting with custom working directory: ${workingDirectoryInput}`);
+    }
+    const workingDirectory = !workingDirectoryInput ? undefined : workingDirectoryInput;
+
+    // move to custom working directory if set
+    if (workingDirectory) {
+      process.chdir(workingDirectory);
```

Comment:
Something seems to be wrong with this call because it [breaks the tests](https://github.com/mxschmitt/action-tmate/pull/157/checks#step:5:187).

The problem may be caused by `process.chdir()` making the main function non-reentrant: the first time it is called with a relative path, it might work, but then the next call will fail because it already changed the working directory to that path.

A better idea may be to teach [the `execShellCommand()` function](https://github.com/mxschmitt/action-tmate/blob/1005f9c9db5f1b055a495e72c6e589764984baf6/src/helpers.js#L19) to accept an (optionally undefined) `cwd` parameter that is passed to [the `spawn()` calls](https://github.com/mxschmitt/action-tmate/blob/1005f9c9db5f1b055a495e72c6e589764984baf6/src/helpers.js#L23-L36), and then specify `workingDirectory` as that `cwd` parameter in [the `new-session` call](https://github.com/mxschmitt/action-tmate/blob/1005f9c9db5f1b055a495e72c6e589764984baf6/src/index.js#L126). That way, the change of the working directory is limited to the call that actually needs it.

Reply:
> A better idea may be to teach [the `execShellCommand()` function](https://github.com/mxschmitt/action-tmate/blob/1005f9c9db5f1b055a495e72c6e589764984baf6/src/helpers.js#L19) to accept an (optionally undefined) `cwd` parameter that is passed to [the `spawn()` calls](https://github.com/mxschmitt/action-tmate/blob/1005f9c9db5f1b055a495e72c6e589764984baf6/src/helpers.js#L23-L36), and then specify `workingDirectory` as that `cwd` parameter in [the `new-session` call](https://github.com/mxschmitt/action-tmate/blob/1005f9c9db5f1b055a495e72c6e589764984baf6/src/index.js#L126).

@chenrui333 This should be _much_ easier now: I already introduced that `options` parameter in https://github.com/mxschmitt/action-tmate/commit/dcd27aa028e970e5f8fd4c8d4796bc2c3b87a1de, and all you need to do is to piggyback on that code by adding `cwd: options && options.cwd` to the options that are passed to `spawn()`.